plugins/plugin-chart-echarts/src/Pie/EchartsPie.tsx

```diff
@@ -16,10 +16,67 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import { EchartsProps } from '../types';
+import { ensureIsArray } from '@superset-ui/core';
+import React, { useState, useEffect } from 'react';
+import { PieChartTransformedProps } from './types';
 import Echart from '../components/Echart';
+import { EventHandlers } from '../types';
 
-export default function EchartsPie({ height, width, echartOptions }: EchartsProps) {
-  return <Echart height={height} width={width} echartOptions={echartOptions} />;
+export default function EchartsPie({
+  height,
+  width,
+  echartOptions,
+  setDataMask,
+  labelMap,
+  groupby,
+}: PieChartTransformedProps) {
+  const [selectedValues, setSelectedValues] = useState<string[]>([]);
```

Comment:
I think it shouldn't be managed here, it should be saved in `currenState.value` and then injected to here from dashboard.
The question that I'm not sure if we need to pass it as `currentValue` of formData, or better pass entire currentState, so here also can be passed also other data